types_linq/enumerable.py

```diff
@@ -274,8 +275,9 @@ def inner():
         return Enumerable(inner)  # type: ignore
 
     def distinct(self) -> Enumerable[TSource_co]:
+        from .util import _Set
```

Comment:
Don't need to import that here. Just import your set at the beginning (like line 11)

I did this for some imports because otherwise they would cause circular ref problem